src/grid/storage.rs

```diff
@@ -232,14 +232,14 @@ impl<T> Storage<T> {
             // Cast to a qword array to opt out of copy restrictions and avoid
             // drop hazards. Byte array is no good here since for whatever
             // reason LLVM won't optimized it.
-            let a_ptr = self.inner.as_mut_ptr().add(a) as *mut u64;
-            let b_ptr = self.inner.as_mut_ptr().add(b) as *mut u64;
+            let a_ptr = self.inner.as_mut_ptr().add(a) as *mut usize;
+            let b_ptr = self.inner.as_mut_ptr().add(b) as *mut usize;
 
             // Copy 1 qword at a time
             //
             // The optimizer unrolls this loop and vectorizes it.
-            let mut tmp: u64;
-            for i in 0..4 {
+            let mut tmp: usize;
+            for i in 0..(::std::mem::size_of::<Row<T>>() as isize / 8) {
```

Comment:
Let's change this back to 4 since we assert above there are 4 `usize` and we are moving `usize` around.